src/Temporalio/Activity/ActivityContext.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.Threading;
+using Google.Protobuf;
+using Microsoft.Extensions.Logging;
+
+namespace Temporalio.Activity
+{
+    /// <summary>
+    /// Context that is available during activity executions. Use <see cref="Current" /> to get the
+    /// context. Contexts are <see cref="AsyncLocal{T}" /> to activities. <see cref="HasCurrent" />
+    /// can be used to check whether a context is available.
+    /// </summary>
+    public class ActivityContext
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ActivityContext"/> class.
+        /// </summary>
+        /// <param name="info">Activity info.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <param name="workerShutdownToken">Workflow shutdown token.</param>
+        /// <param name="taskToken">Raw activity task token.</param>
+        /// <param name="logger">Logger.</param>
+#pragma warning disable CA1068 // We don't require cancellation token as last param
+        internal ActivityContext(
+            ActivityInfo info,
+            CancellationToken cancellationToken,
+            CancellationToken workerShutdownToken,
+            ByteString taskToken,
+            ILogger logger)
+        {
+            Info = info;
+            CancellationToken = cancellationToken;
+            WorkerShutdownToken = workerShutdownToken;
+            TaskToken = taskToken;
+            Logger = logger;
+        }
+#pragma warning restore CA1068
+
+        /// <summary>
+        /// Gets a value indicating whether the current code is running in an activity.
+        /// </summary>
+        public static bool HasCurrent => AsyncLocalCurrent.Value != null;
+
+        /// <summary>
+        /// Gets the current activity context.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">If no context is available.</exception>
+        public static ActivityContext Current => AsyncLocalCurrent.Value ??
+            throw new InvalidOperationException("No current context");
+
+        /// <summary>
+        /// Gets the info for this activity.
+        /// </summary>
+        public ActivityInfo Info { get; private init; }
+
+        /// <summary>
+        /// Gets the logger scoped to this activity.
+        /// </summary>
+        public ILogger Logger { get; private init; }
+
+        /// <summary>
+        /// Gets why the activity was cancelled. This value is inaccurate until
+        /// <see cref="CancellationToken" /> is cancelled.
+        /// </summary>
+        public ActivityCancelReason CancelReason => CancelReasonRef.CancelReason;
```

Comment:
:+1: Will change to `None` per https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/enum. I originally didn't want to use `None` because it implies that it wasn't cancelled, but rereading it just says there's no reason.

Reply:
Fixed